src/drivers/mct/cime_config/buildnml

```diff
@@ -171,6 +171,22 @@ def _create_drv_namelists(case, infile, confdir, nmlgen, files):
         nmlgen.set_value('esp_cpl_dt', value=esp_time)
     # End if pause is active
 
+    # Sanity checks on mem-prof options
+    info_mprof = case.get_value('INFO_MPROF')
+    expect(info_mprof in [0,1,2,3,4],
```

Comment:
There is no need for this check, the <valid_values> field from xml is already checked when you try to change a value using xmlchange.